juju/juju
# [JUJU-344] Wires up the uniter for caas probe support.

The uniter now implements probes that are apart of the /startup probe
support in Kubernetes. This PR add this probe into the list of probes
checked.

PR also introduces a new detailed mode for probe endpoints that can
specifically list which probes are passing and failing.

## Checklist


## QA steps

1. Bootstrap Juju to microk8s
2. Deploy snappass-test
3. Confirm the startup probes are working with:
```
kubectl port-forward snappass-test-0 -n controller-microk8s-localhost 3856:3856
curl http://localhost:3856/startup?detailed=true
```

## Documentation changes

N/A

## Bug reference

N/A
